Fixes buggy line highlighting

cctechwiz opened the pull request:
**Your checklist for this pull request**
- [X] I've read the [guidelines for contributing](https://cutter.re/docs/contributing/code/getting-started.html) to this repository
- [X] I made sure to follow the project's [coding style](https://cutter.re/docs/contributing/code/development-guidelines.html)
- [X] I've updated the [documentation](https://cutter.re/docs/user-docs.html) with the relevant information (if needed)


**Detailed description**

This PR solves two related issues:

1. The currently selected line would no longer be highlighted if it was the very last line in the window.
2. The PC line would no longer be highlighted if the currently selected line was off the screen.

The solutions for each are as follows:

1. Calling `updateCursorPosition` before moving the cursor. Previously the call to `readCurrentDisassemblyOffset` in `updateCursorPosition` was causing essentially an off-by-one bug since the cursor was moved prior to checking the current offset.
2. Separated `highlightCurrentLine` and `highlightPCLine` logic so they can be called independently when needed. Previously logic for highlighting the PC was included in highlighting the current line. This caused the PC to not be highlighted when the current line was not on-screen and being highlighted.


**Test plan (required)**
See the steps detailed in #2317 for how to recreate the issue. Following the steps in the ticket no longer cause the bugs in question. Below are screenshots similar to those in the original issue, but this time showing the desired functionality is working. The steps mention refer to the recreation steps from the issue.

Step 1 - 2. The green line indicates current PC and gray/blue is selected line.
![1and2](https://user-images.githubusercontent.com/8186200/95666784-6d8dd900-0b1a-11eb-9048-b4fd128a26fd.PNG)

Step 3, correct highlighting now
![3](https://user-images.githubusercontent.com/8186200/95666813-b8a7ec00-0b1a-11eb-9d19-02c4d1801aa8.PNG)

Step 4, highlighting is now present
![4](https://user-images.githubusercontent.com/8186200/95666815-bf366380-0b1a-11eb-8495-556bf41da853.PNG)


**Closing issues**
closes #2317

ITAYC0HEN commented on src/widgets/DisassemblyWidget.h:
```diff
@@ -29,6 +29,7 @@ class DisassemblyWidget : public MemoryDockWidget
 
 public slots:
     void highlightCurrentLine();
+    void highlightPCLine();
```
Do you mind documenting these two functions for future reference? :)
We are using the Doxygen docstring format: https://cutter.re/docs/contributing/code/development-guidelines.html#docstrings

cctechwiz replied:
Will do.

ITAYC0HEN reviewed (approved):
Welcome @cctechwiz and thank you for your contribution!
Your pull request looks great, and I liked the process you took to investigate the issue and explain it to us. That's very nice :)

The PR looks good to me, I only left on request for adding docstrings, that's it.
Would love @karliss review to make sure the issue he experienced was indeed solved.

Thanks again and looking forward to your next contribution :) Let us know if you want any guideline finding your next target :sunglasses:

karliss commented on src/widgets/DisassemblyWidget.cpp:
```diff
@@ -420,8 +428,9 @@ void DisassemblyWidget::highlightCurrentLine()
-                extraSelections.append(highlightSelection);
-            } else if (lineOffset != RVA_INVALID && lineOffset > PCAddr) {
+                pcSelections.append(highlightSelection);
+            }
+            else if (lineOffset != RVA_INVALID && lineOffset > PCAddr) {
```
Since Cutter [coding style](https://cutter.re/docs/contributing/code/development-guidelines.html) places braces on the same line as if and else, I would say this should be on the same line.
```
if () {
} else if () {
}
```
instead of 
```
if () {
}
else if () {
}
```

karliss reviewed (changes requested):
Two tiny suggestions, otherwise looks good.

As a general advice for contributing to Cutter and other projects on GitHub: when working on your PR it's better to create a separate branch for each PR instead of comiting directly to master of your fork.  That way you don't have any problems if you decide to work on a second PR while first one is being reviewed and you don't have to think afterwards how to cleanup the master branch after the PR is done. You don't need to change anything right now for this PR, but keep this in mind for your future contributions.

You can read a bit more about it here https://git-scm.com/book/en/v2/GitHub-Contributing-to-a-Project .

cctechwiz commented:
Doxygen comments have been added.

Fixed the manual appending to the correct way to append QLists together.

Fixed braces to match code style (I think VSCode autoformatted that function after I copied it once...).

About using the master branch. I created a feature branch and everything but I guess I forgot to switch to it and didn't notice till I was making the PR and didn't feel like fixing it at that point. Thank you for the guidance.

yossizap reviewed (approved):
LGTM, tested in Windows 10. It's good to see new debugger contributions :)
